compiler/dyno/lib/uast/AnonFormal.cpp

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2021-2022 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "chpl/uast/AnonFormal.h"
+
+#include "chpl/uast/Builder.h"
+
+namespace chpl {
+namespace uast {
+
+
+owned<AnonFormal>
+AnonFormal::build(Builder* builder, Location loc,
+                  AnonFormal::Intent intent,
+                  owned<AstNode> typeExpression) {
+  AstList children;
+  int8_t typeExpressionChildNum = AstNode::NO_CHILD;
+
+  if (typeExpression.get() != nullptr) {
```

Comment:
I think `std::unique_ptr` has an overload for `!=` and `nullptr_t`:

```C++
template <class T, class D> bool operator!= (const unique_ptr<T,D>& lhs, nullptr_t) noexcept;
```

So you might not need the `.get()` here (and maybe other places that need to compare against null)

Reply:
Ah, ok, ty!